nin/frontend/app/scripts/controllers/main.js

```diff
@@ -85,6 +85,23 @@
           ]
         },
         {
+          name: 'Generate',
+          items: [
+            {name: 'Layer', click: function() {
+              var layerName = window.prompt("Enter a name for the layer:");
```

Comment:
Prompt blocks js, but does not pause audio. This means that if the demo is running while somebody triggers this menu action, a potentially large number of frames need to be updated and maybe rendered for the js to catch up with the music once the prompt disappears. This can hang the browser, if you are unlucky.